### ISSUE TITLE
"create_rse" error: "Error in match.arg(project_home) "

### ISSUE THREAD

**Issue by csb-gz** [LieberInstitute/recount3#6]
The  create_rse works very nicely for most projects. However, I recently discovered a strange case for "SRP102119" dataset. I got an error about "Error in match.arg(project_home) :    'arg' should be one of “data_sources/sra”, “data_sources/gtex”, “data_sources/tcga”" . Could you help take a look? Many thanks. 
```
human_projects <- available_projects()
query_project <- subset(
    human_projects,
    project=="SRP102119" & project_type == "data_sources"
)
rse_gene_SRP102119 <- create_rse(query_project,type="gene",annotation="gencode_v29")

```

**Comment by lcolladotor** [comment]
Hi,

I can reproduce your problem with:

```R
library("recount3")
human_projects <- available_projects()
query_project <- subset(
    human_projects,
    project=="SRP102119" & project_type == "data_sources"
)
query_project
#        project organism file_source     project_home project_type n_samples
# 6091 SRP102119    human         sra data_sources/sra data_sources        37
rse_gene_SRP102119 <- create_rse(query_project,type="gene",annotation="gencode_v29")
traceback()
rse_gene_SRP102119
sessionInfo()
```

<details>


```R
> library("recount3")
Loading required package: SummarizedExperiment
Loading required package: MatrixGenerics
Loading required package: matrixStats

Attaching package: ‘MatrixGenerics’

The following objects are masked from ‘package:matrixStats’:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars

Loading required package: GenomicRanges
Loading required package: stats4
Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors

Attaching package: ‘S4Vectors’

The following objects are masked from ‘package:base’:

    expand.grid, I, unname

Loading required package: IRanges
Loading required package: GenomeInfoDb
Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.


Attaching package: ‘Biobase’

The following object is masked from ‘package:MatrixGenerics’:

    rowMedians

The following objects are masked from ‘package:matrixStats’:

    anyMissing, rowMedians

> human_projects <- available_projects()
2021-08-02 23:20:03 caching file sra.recount_project.MD.gz.
2021-08-02 23:20:08 caching file gtex.recount_project.MD.gz.
2021-08-02 23:20:12 caching file tcga.recount_project.MD.gz.
> query_project <- subset(
+     human_projects,
+     project=="SRP102119" & project_type == "data_sources"
+ )
> query_project
       project organism file_source     project_home project_type n_samples
6091 SRP102119    human         sra data_sources/sra data_sources        37
> rse_gene_SRP102119 <- create_rse(query_project,type="gene",annotation="gencode_v29")
2021-08-02 23:21:04 downloading and reading the metadata.
2021-08-02 23:21:05 caching file sra.sra.SRP102119.MD.gz.
adding rname 'http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/19/SRP102119/sra.sra.SRP102119.MD.gz'
  |======================================================================| 100%

2021-08-02 23:21:10 caching file sra.recount_project.SRP102119.MD.gz.
adding rname 'http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/19/SRP102119/sra.recount_project.SRP102119.MD.gz'
  |======================================================================| 100%

2021-08-02 23:21:15 caching file sra.recount_qc.SRP102119.MD.gz.
adding rname 'http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/19/SRP102119/sra.recount_qc.SRP102119.MD.gz'
  |======================================================================| 100%

2021-08-02 23:21:20 caching file sra.recount_seq_qc.SRP102119.MD.gz.
adding rname 'http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/19/SRP102119/sra.recount_seq_qc.SRP102119.MD.gz'
  |======================================================================| 100%

2021-08-02 23:21:25 caching file sra.recount_pred.SRP102119.MD.gz.
adding rname 'http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/19/SRP102119/sra.recount_pred.SRP102119.MD.gz'
  |======================================================================| 100%

Error in match.arg(project_home) :
  'arg' should be one of “data_sources/sra”, “data_sources/gtex”, “data_sources/tcga”
> traceback()
5: stop(gettextf("'arg' should be one of %s", paste(dQuote(choices),
       collapse = ", ")), domain = NA)
4: match.arg(project_home)
3: locate_url(project = project, project_home = project_home, type = "bw",
       organism = organism, annotation = annotation, recount3_url = recount3_url,
       sample = metadata$external_id)
2: create_rse_manual(project = project_info$project, project_home = project_info$project_home,
       type = type, organism = project_info$organism, annotation = annotation,
       bfc = bfc, jxn_format = jxn_format, recount3_url = recount3_url,
       verbose = verbose)
1: create_rse(query_project, type = "gene", annotation = "gencode_v29")
> sessionInfo()
R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets
[8] methods   base

other attached packages:
 [1] recount3_1.2.1              SummarizedExperiment_1.22.0
 [3] Biobase_2.52.0              GenomicRanges_1.44.0
 [5] GenomeInfoDb_1.28.1         IRanges_2.26.0
 [7] S4Vectors_0.30.0            BiocGenerics_0.38.0
 [9] MatrixGenerics_1.4.0        matrixStats_0.60.0
[11] testthat_3.0.4              devtools_2.4.2
[13] usethis_2.0.1

loaded via a namespace (and not attached):
 [1] httr_1.4.2               pkgload_1.2.1            bit64_4.0.5
 [4] R.utils_2.10.1           assertthat_0.2.1         BiocFileCache_2.0.0
 [7] blob_1.2.2               GenomeInfoDbData_1.2.6   Rsamtools_2.8.0
[10] yaml_2.2.1               remotes_2.4.0            sessioninfo_1.1.1
[13] pillar_1.6.2             RSQLite_2.2.7            lattice_0.20-44
[16] glue_1.4.2               XVector_0.32.0           Matrix_1.3-4
[19] R.oo_1.24.0              XML_3.99-0.6             pkgconfig_2.0.3
[22] zlibbioc_1.38.0          purrr_0.3.4              processx_3.5.2
[25] BiocParallel_1.26.1      tibble_3.1.3             generics_0.1.0
[28] ellipsis_0.3.2           cachem_1.0.5             withr_2.4.2
[31] cli_3.0.1                magrittr_2.0.1           crayon_1.4.1
[34] memoise_2.0.0            ps_1.6.0                 R.methodsS3_1.8.1
[37] fs_1.5.0                 fansi_0.5.0              pkgbuild_1.2.0
[40] tools_4.1.0              data.table_1.14.0        prettyunits_1.1.1
[43] BiocIO_1.2.0             lifecycle_1.0.0          DelayedArray_0.18.0
[46] callr_3.7.0              Biostrings_2.60.1        compiler_4.1.0
[49] rsthemes_0.2.1.9000      colorout_1.2-2           rlang_0.4.11
[52] grid_4.1.0               RCurl_1.98-1.3           rstudioapi_0.13
[55] rjson_0.2.20             rappdirs_0.3.3           prompt_1.0.1
[58] bitops_1.0-7             restfulr_0.0.13          DBI_1.1.1
[61] curl_4.3.2               R6_2.5.0                 GenomicAlignments_1.28.0
[64] dplyr_1.0.7              rtracklayer_1.52.0       fastmap_1.1.0
[67] bit_4.0.4                utf8_1.2.2               filelock_1.0.2
[70] rprojroot_2.0.2          desc_1.3.0               Rcpp_1.0.7
[73] vctrs_0.3.8              dbplyr_2.1.1             tidyselect_1.1.1

> query_project$project_home
[1] "data_sources/sra"
```

</details>

Hmm... I'll have to dive deeper into this.

Best,
Leo

**Comment by lcolladotor** [comment]
Hi,

Actually this issue got resolved with #5.

```R
master > rse_gene_SRP102119 <- create_rse(query_project,type="gene",annotation="gencode_v29")
2021-08-03 07:21:46 downloading and reading the metadata.
2021-08-03 07:21:48 caching file sra.sra.SRP102119.MD.gz.
2021-08-03 07:21:51 caching file sra.recount_project.SRP102119.MD.gz.
2021-08-03 07:21:54 caching file sra.recount_qc.SRP102119.MD.gz.
2021-08-03 07:21:57 caching file sra.recount_seq_qc.SRP102119.MD.gz.
2021-08-03 07:22:00 caching file sra.recount_pred.SRP102119.MD.gz.
2021-08-03 07:22:03 downloading and reading the feature information.
2021-08-03 07:22:04 caching file human.gene_sums.G029.gtf.gz.
adding rname 'http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G029.gtf.gz'
  |=================================================================================================================================================================================================| 100%

2021-08-03 07:22:10 downloading and reading the counts: 37 samples across 64837 features.
2021-08-03 07:22:11 caching file sra.gene_sums.SRP102119.G029.gz.
adding rname 'http://duffel.rail.bio/recount3/human/data_sources/sra/gene_sums/19/SRP102119/sra.gene_sums.SRP102119.G029.gz'
  |=================================================================================================================================================================================================| 100%

2021-08-03 07:22:17 construcing the RangedSummarizedExperiment (rse) object.
Warning message:
The following metadata files are empty and will be dropped: sra.recount_pred.SRP102119.MD.gz 
master > rse_gene_SRP102119
class: RangedSummarizedExperiment 
dim: 64837 37 
metadata(8): time_created recount3_version ... annotation recount3_url
assays(1): raw_counts
rownames(64837): ENSG00000278704.1 ENSG00000277400.1 ... ENSG00000182484.15_PAR_Y ENSG00000227159.8_PAR_Y
rowData names(10): source type ... havana_gene tag
colnames(37): SRR5357745 SRR5357746 ... SRR5357780 SRR5357781
colData names(168): rail_id external_id ... recount_seq_qc.errq BigWigURL
master > packageVersion("recount3")
[1] ‘1.3.2’
```

Use `recount3` version 1.2.2 or 1.3.2 from bioc-release or bioc-devel. They should be available from Bioconductor within 48-72 hours.

Best,
Leo

**Comment by csb-gz** [comment]
Thanks!
